client/network/test/src/lib.rs

```diff
@@ -971,80 +975,91 @@ where
 		self.rt_handle().spawn(f);
 	}
 
-	/// Polls the testnet until all nodes are in sync.
+	/// Polls the testnet until all peers are connected to each other.
 	///
 	/// Must be executed in a task context.
-	fn poll_until_sync(&mut self, cx: &mut FutureContext) -> Poll<()> {
+	fn poll_until_connected(&mut self, cx: &mut FutureContext) -> Poll<()> {
 		self.poll(cx);
 
-		// Return `NotReady` if there's a mismatch in the highest block number.
+		let num_peers = self.peers().len();
+		if self.peers().iter().all(|p| p.num_peers() == num_peers - 1) {
+			return Poll::Ready(())
+		}
+
+		Poll::Pending
+	}
+
+	async fn is_in_sync(&mut self) -> bool {
 		let mut highest = None;
-		for peer in self.peers().iter() {
-			if peer.is_major_syncing() || peer.network.num_queued_blocks() != 0 {
-				return Poll::Pending
+		let peers = self.peers_mut();
+
+		for peer in peers {
+			if peer.sync_service.is_major_syncing() ||
+				peer.sync_service.num_queued_blocks().await.unwrap() != 0
```

Comment:
is it guaranteed to always return Some(n)?

Reply:
Nope but these are tests so I'm not that concerned with `unwrap()`